### DISCUSSION TITLE
413 Request Entity Too Large

### DISCUSSION THREAD

**Discussion by maccuryj** [community/community#62913] [Codespaces]
### Select Topic Area

Bug

### Body

I am currently testing a Python Flask API in Codespaces aimed at accepting .mp4 files. At some point, my endpoint stopped working, yielding *413 Request Entity Too Large* errors. This response does not come from my backend. Looking at the cURL log, it seems codespaces has restricted the size limit for request data:

```
< HTTP/1.1 100 Continue
* We are completely uploaded and fine
< HTTP/1.1 413 Request Entity Too Large
< Date: Sun, 06 Aug 2023 14:03:21 GMT
< Connection: Keep-Alive
< X-Served-By: tunnels-prod-rel-euw-v3-cluster
< Content-Type: text/html
< Content-Length: 176
< Strict-Transport-Security: max-age=31536000; includeSubDomains
<
<html>
<head><title>413 Request Entity Too Large</title></head>
<body>
<center><h1>413 Request Entity Too Large</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

I was not aware of any such limitations and am not able to find corresponding documentation.
This problem has only started arising in the past days, it worked as expected ~2 weeks ago.
Will this limit persist on my account?

Thank you.

**Comment by Dev0psNewb** [comment]
same issue here

**Comment by jkeech** [comment]
@maccuryj how large are the .mp4 files you are uploading? I believe there's a 16MB limit on the HTTP request, although this is not clearly documented currently. If your file is over 16MB, that would explain why it's failing. We can look at updating the docs.

One workaround might be to connect from VS Code desktop and use local port forwarding over a `localhost` url rather than going through the port forwarding endpoint at `https://<codespace-name>-<port>.app.github.dev`.

**Reply by Dev0psNewb** [reply]
I'm seeing this for files well below 16MB, and it says the restriction is being enforced by NginX.
So I bet an update came along introducing this right about the time where the format of the Codespaces URL's changed too. (and messed up the captcha's domain reg's too)

**Reply by jkeech** [reply]
@Dev0psNewb can you provide a sample repo with steps to reproduce the problem?

**Reply by Dev0psNewb** [reply]
yup: https://github.com/Dev0psNewb/413-POC

**Reply by jkeech** [reply]
Thanks! I'm able to confirm that 1 MB is where it's erroring out. It succeeds with a file that is 1023KB and fails with 1024KB (1MB). We're investigating.

**Reply by Dev0psNewb** [reply]
Excellent :-)

So the solution for your NginX prolly is a change to conf.d, something along the lines: client_max_body_size 25M;
(Or whatever size you find usefull)

**Reply by jkeech** [reply]
This should be fixed now. We've increased the limit to 16M.

**Reply by Dev0psNewb** [reply]
Yes, I can confirm this too on my "production" codespace site. Thanks for responding to this thread, Awesome work mate !

**Reply by MGackstatter** [reply]
@jkeech can we upload >16MB files in codespaces, e.g. using Github LFS? Would appreciate pointers!
